plugins/RedditPlugin/__init__.py

```diff
@@ -0,0 +1,79 @@
+from GarfieldBot import GarfieldPlugin, MessageEvent
+from dotenv import load_dotenv
+import praw
+import os
+
+class RedditPlugin(GarfieldPlugin):
+    """ Get top 5 articles from a given subreddit and display 5 comments from the articles """
+
+    def __init__(self, manifest, bot):
+        super().__init__(manifest, bot)
+        
+        # Get login details from env
+        load_dotenv()
+        self.reddit_client_id=os.environ["REDDIT_CLIENT_ID"]
+        self.reddit_client_secret=os.environ["REDDIT_CLIENT_SECRET"]
+        self.reddit_client_username=os.environ["REDDIT_USERNAME"]
+        self.reddit_client_password=os.environ["REDDIT_PASSWORD"]
+        
+        # Login
+        self.reddit = praw.Reddit(
+            client_id       =self.reddit_client_id,
+            client_secret   =self.reddit_client_secret,
+            username        =self.reddit_client_username,
+            password        =self.reddit_client_password,
+            user_agent      ='GarfieldBotRedditPlugin'
+        )
+
+        # Register commands
+        self.bot.register_command("reddit", self.handle_reddit)
+        self.bot.register_command("comments", self.handle_comments)
+
+    def handle_reddit(self, event: MessageEvent, *args) -> None:
+        # Create a reference to the subreddit specified
+        subreddit = self.reddit.subreddit(args[0])
+        
+        # Get the top 5 submissions of the day
+        links = 1
+        for submission in subreddit.top('day'):
+            # Skip all submissions tagged NSFW
+            if(submission.over_18):
+                continue
+            
+            # Build the message for the channel
+            message = "!comments " + str(submission.id) + " *" + str(submission.title) + "*\n" + str(submission.url)
```

Comment:
Rather than using recursive command calls, actually add the thread to the message when you send it.

Reply:
Delegated to @lavenduale